firebase-admin/runtime/src/main/java/io/quarkiverse/googlecloudservices/firebase/admin/runtime/authentication/http/FirebaseSecurityAuthMechanism.java

```diff
@@ -56,11 +67,18 @@ public Uni<SecurityIdentity> authenticate(RoutingContext context, IdentityProvid
                     String tokenChallenge = current.substring(PREFIX_LENGTH);
 
                     // We have found a suitable header, so try to authenticate
-                    return identityProviderManager.authenticate(new FirebaseAuthenticationRequest(tokenChallenge));
+                    return identityProviderManager.authenticate(new FirebaseAuthenticationRequest(tokenChallenge, false));
                 }
             }
         }
 
+        if (sessionCookiesEnabled) {
+            var manager = cookieManagerInstance.get();
+            var cookie = manager.getSessionCookie(context);
+            return cookie.map(c -> identityProviderManager
+                    .authenticate(new FirebaseAuthenticationRequest(c, true))).orElse(Uni.createFrom().nullItem());
```

Comment:
@sberyozkin This is where the cookies are validated retrieved from the manager and authenticated using the identityProviderManager